py3status/modules/xscreensaver.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""
+Toggle and lock Xscreensaver.
+
+This is a partial rewrite of dpms.py. Most credit goes to the original author
+Andre Doser. This module shows and controls the status of Xscreensaver
+
+This script is useful for people who let Xscreensaver manage dpms settings.
+Xscreensaver has its own dpms variables separate from xset. Dpms can be
+safely turned off in xset as long as xscreensaver is running.
+Settings can be managed using "xscreensaver-demo".
+
+Configuration parameters:
+    button_lock: mouse button to lock Xscreensaver (default 3)
+    button_toggle: mouse button to toggle Xscreensaver (default 1)
+    cache_timeout: refresh interval for this module (default 15)
+    format: default format for this module (default '{icon}')
+    icon_off: display when Xscreensaver is not running (default 'XSCR')
+    icon_on: display when Xscreensaver is running (default 'XSCR')
+
+ Format placeholders:
+    {icon} display current xscreensaver icon
+
+@author neutronst4r <c7420{at}gmx{dot}net>
+"""
+from os import setpgrp
+from subprocess import Popen, DEVNULL
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_lock = 3
+    button_toggle = 1
+    cache_timeout = 15
+    format = '{icon}'
+    icon_off = 'XSCR'
+    icon_on = 'XSCR'
+
+    def post_config_hook(self):
+        if not self.py3.check_commands(['xscreensaver']):
+            self.py3.notify_user('xscreensaver is not installed. cannot use this module.')
+
+    def xscreensaver(self):
+        """
+        """
+        try:
+            run = self.py3.command_output(['pidof', 'xscreensaver'])
+        except:
+            run = False
+
```

Comment:
As @tobes has suggested saving the result of run is kind of unnecessary. We could do something like:
```
def _is_running_(self):
       """
       tests whether xscreensaver is running, returns boolean
       """
       try:
           self.py3.command_output(['pidof', 'xscreensaver'])
       except:
           return False
       else:
           return True
```
and ask if xscreensaver is running.